server/routes/events.js

```diff
@@ -1,15 +1,258 @@
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	});
+  } else if (year != undefined) {
+  	db.query('SELECT * FROM event WHERE EXTRACT(YEAR FROM starts_at) = \'' + year + '\' OR EXTRACT(YEAR FROM ends_at) = \'' + year + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (quarter != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE quarter = \'' + quarter + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (start_date != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE date(starts_at) = \'' + start_date + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (end_date != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE date(starts_at) = \'' + end_date + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (category != undefined) {
+  	db.query('SELECT event_id FROM event_category WHERE category_id = \'' + category_id + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (company != undefined) {
+  	db.query('SELECT event_id FROM event_company WHERE company_id = \'' + company + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (featured != undefined) {
+  	db.query('SELECT * FROM event WHERE is_featured = \'' + featured + '\'', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else if (past != undefined) {
+  	db.query('SELECT fb_id FROM event WHERE ends_at < now()', [], (err, result) => {
+  		if (err) {
+  			return next(err);
+  		}
+  		res.send(result.rows);
+  	})
+  } else {
+  	res.json({result: false}); // error message
+  }
 });
 
-// GET events listing. 
+// Get all events
 router.get('/', function(req, res, next) {
-    // TODO
+  db.query('SELECT * FROM event', [], (err, result) => {
+    	if (err) {
+      	  return next(err);
+    	}
+    	res.send(result.rows);
+  	});
+});
+
+// Get all event names
+// NOTE: chose to SELECT fb_id instead of name
+router.get('/names', function(req, res, next) {
+  db.query('SELECT fb_id FROM event', [], (err, result) => {
+    	if (err) {
+      	  return next(err);
+    	}
+    	res.send(result.rows);
+  	});
+});
+
+//Get all events containing term substring
+//I think this needs to go before all of the /:event_id, or else it interprets search as an event id :/
+//also I can't figure out how to use the [] for this one :(
+router.get('/search', function(req, res, next) {
+	const term = req.query.term;
```

Comment:
This indentation seems kinda funky too

Reply:
Also any query parameters should be validated before use (check that they exist/are valid values)